torch/_inductor/index_propagation.py

```diff
@@ -322,6 +323,13 @@ def indirect_indexing(
 
             expr = sympy.sympify(index.value.expr)
 
+            if free_symbol_is_type(expr, SymT.INDIRECT):
+                # This is the nested indirect indexing case:
+                # a = ops.indirect_indexing(...)
+                # b = ops.index_expr(a, ...)
+                # c = ops.indirect_indexing(b, ...)
```

Comment:
In this case, it doesn't seem like we can meaningfully propagate a sympy expression, so don't we want to fallback?

Reply:
No, we should be able to propagate the sympy expression without issue. We should just fix the `_maybe_evaluate_static` call so it can handle `indirect0`.

Ideally we would keep track of the bounds for `ops.indirect_indexing` since these do have known bounds. But at the very least we should treat it as unbounded.

Reply:
For example
```
a = ops.indirect_indexing(...)
b = ops.index_expr(a, torch.int32)
c = ops.add(b, ops.constant(1, torch.int32))
d = ops.indirect_indexing(c, ...)
```

`d = indirect1` even though we know that `indirect1 = indirect0 + 1`

We can simplify to:
```
a = ops.indirect_indexing(...)
d = a + 1
```

where `d = indirect0 + 1`.

Reply:
To do this we should:
- Make sure that the shapes of the tow indirect_indexings are the same
- Make sure we are not wrapping the variable twice (I think we may be currently wrapping it twice).

It's not clear to me whether we can do the first thing in the IndexPropagation pass tho.

Reply:
We should propagate the bounds from the first `indirect0` into the reasoning, so we might be able to elide the wrapping.

In general though, we might need to have two wrappings and two different sizes because each `ops.indirect_indexing` call might have come from a different tensor which may have different shapes and each are wrapped independently.

Reply:
indirect0 variables barely ever have bounds, as loads do not have bounds, and indirect indexing is often performed on information that comes from a load.

But yeah, I guess that in general it makes sense for the first indirect to be wrapped and so on at a codegen level, and then we can have the second indirect wrapped here.

Reply:
> indirect0 variables barely ever have bounds,

On the contrary, `indirect0` always has a bound. `ops.indirect_indexing(var, size)` has the bound [0, size) as checked by the runtime assert.

Reply:
ugh, right.